### PR TITLE
Lets express time complexity as well.

### PR DESCRIPTION
**Pull request by way2datta**
> What was wrong/why is this fix needed? (quick summary only)

> Issue number (if there is an associated issue)

> Anything else that could help us review it

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/array/concat/index.html [hunk 1]
```diff
@@ -16,6 +16,8 @@
 <p>The <code><strong>concat()</strong></code> method is used to merge two or more arrays.
   This method does not change the existing arrays, but instead returns a new array.</p>
 
+This method is a linear time operation, taking O(n) time.
```
Comment (peterbe) [inline files/en-us/web/javascript/reference/global_objects/array/concat/index.html:19]:
```suggestion
<p>This method is a linear time operation, taking <code>O(n)</code> time.</p>
```

**Comment by teoli2003** [comment]
I find having complexity information useful, but is it required by the spec, or can we have different engines implementing this feature with different complexities?

**Comment by wbamberg** [comment]
Also, if we do want to include this, it should be in the Description, not right after the summary.

**Comment by sideshowbarker** [comment]
For `concat()` there is no specific algorithm in the ES spec — and otherwise no specific requirements in the ES spec — which would lead to be able to state unequivocally that `concat()` is going to be linear time. To put it in other words, it’s easily imaginable that there are an unbounded number of ways that an engine could choose to implement an algorithm for `concat()` that would have much worse performance than linear time, while still conforming the requirements in the spec.

I guess it might be possible to say that `concat()` *can* be implemented in linear time — but I’m not even sure we can say that it would always be linear time for all cases.

Regardless, I am not aware of any other places where we attempt to state anything about the Big O performance characteristics of implementations of particular JavaScript methods. So there is no apparent good reason to be stating it just for this one method.

Given all that, I’m going ahead and closing this PR. If someone comes forward with new information, then of course it could be revisited.

**Pull request closed without merging**